ipa-core/src/ff/prime_field.rs

```diff
@@ -163,9 +163,15 @@ macro_rules! field_impl {
             type Output = Self;
 
             fn mul(self, rhs: Self) -> Self::Output {
-                debug_assert!(u32::try_from(Self::PRIME).is_ok());
-                let c = u64::from;
+                debug_assert!(<$store>::try_from(Self::PRIME).is_ok());
+                // check container for multiply is large enough
+                debug_assert!(
+                    <$store_multiply>::MAX >> $bits
+                        >= <$store_multiply>::try_from(<$store>::MAX).unwrap()
+                );
```

Comment:
Andy recently added `const_assert!` macro that can be useful here.

Reply:
I cannot use try_from since it is non_constant. I used as u128 instead to make const_assert! work. The assert only works as long as we dont use larger (or not unsigned) types than u128.